core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -545,6 +559,50 @@ default void finishUpdate(ConnectorSession session, ConnectorTableHandle tableHa
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support updates");
     }
 
+    /**
+     * Return the row update paradigm supported by the connector on the table, or
+     * {@link RowChangeParadigm#ROW_CHANGE_NOT_SUPPORTED} if row change is not supported.
+     * @param session The session in which to start the update operation.
+     * @param tableHandle A ConnectorTableHandle for the table to be updated.
+     * @return Return the supported row update paradigm.
+     */
+    default RowChangeParadigm getRowChangeParadigm(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return ROW_CHANGE_NOT_SUPPORTED;
```

Comment:
is ROW_CHANGE_NOT_SUPPORTED a valid option?

it doesn't seem like it is, so perhaps throw here, and remove that option

Reply:
Agreed that for the default method, throwing is better.  Changed.